File: app/src/main/java/framework/data/Constant.java
package framework.data;


import develop.y.zhzl.R;
import framework.utils.UIUtils;

/**
 * by y on 2016/8/7.
 */
public class Constant {

    public static final int RECYCLERVIEW_LISTVIEW = 1;
    public static final int RECYCLERVIEW_GRIDVIEW = 5;

    public static final String BITMAP_CACHE_KEY = "key";
    public static final String ANNAL_TAG = "annal";
    public static final String THEME_TAG = "theme";

    public static final int LIMIT = 20;
    public static final int OFFSET = 0;

    public static final String ZHIHU = "zhihu";
    public static final String MOVIE = "movie";
    public static final String MUSIC = "music";
    public static final String DEVELOP = "develop";
    public static final String BOOK = "book";
    public static final String INTERNET = "internet";

    public static final boolean DAY = true;
    public static final boolean NIGHT = false;


    public static final int DAY_STYLES = R.style.AppTheme_Day;
    public static final int NIGHT_STYLES = R.style.AppTheme_Night;

    public static String[] getTabName(String type) {
        switch (type) {
            case Constant.ZHIHU:
                return UIUtils.getStringArray(R.array.zhihu);
            case Constant.MOVIE:
                return UIUtils.getStringArray(R.array.movie);
            case Constant.MUSIC:
                return UIUtils.getStringArray(R.array.music);
            case Constant.DEVELOP:
                return UIUtils.getStringArray(R.array.develop);
            case Constant.BOOK:
                return UIUtils.getStringArray(R.array.book);
            case Constant.INTERNET:
                return UIUtils.getStringArray(R.array.internet);
        }
        return new String[0];
    }


    public static String getSuffix(int position, String type) {
        switch (type) {
            case Constant.ZHIHU:
                return UIUtils.getStringArray(R.array.zhihu_suffix)[position];
            case Constant.MOVIE:
                return UIUtils.getStringArray(R.array.movie_suffix)[position];
            case Constant.MUSIC:
                return UIUtils.getStringArray(R.array.music_suffix)[position];
            case Constant.DEVELOP:
                return UIUtils.getStringArray(R.array.develop_suffix)[position];
            case Constant.BOOK:
                return UIUtils.getStringArray(R.array.book_suffix)[position];
            case Constant.INTERNET:
                return UIUtils.getStringArray(R.array.internet_suffix)[position];
            default:
                return "";
        }
    }
}

File: app/src/main/java/framework/data/Constant.kt
package framework.data


import develop.y.zhzl.R
import framework.utils.UIUtils

/**
 * by y on 2016/8/7.
 */
object Constant {

    val RECYCLERVIEW_LISTVIEW = 1
    val RECYCLERVIEW_GRIDVIEW = 5

    val BITMAP_CACHE_KEY = "key"
    val ANNAL_TAG = "annal"
    val THEME_TAG = "theme"

    val LIMIT = 20
    val OFFSET = 0

    val ZHIHU = "zhihu"
    val MOVIE = "movie"
    val MUSIC = "music"
    val DEVELOP = "develop"
    val BOOK = "book"
    val INTERNET = "internet"

    val DAY_STYLES = R.style.AppTheme_Day
    val NIGHT_STYLES = R.style.AppTheme_Night

    fun getTabName(type: String): Array<String> {
        when (type) {
            Constant.ZHIHU -> return UIUtils.getStringArray(R.array.zhihu)
            Constant.MOVIE -> return UIUtils.getStringArray(R.array.movie)
            Constant.MUSIC -> return UIUtils.getStringArray(R.array.music)
            Constant.DEVELOP -> return UIUtils.getStringArray(R.array.develop)
            Constant.BOOK -> return UIUtils.getStringArray(R.array.book)
            else -> return UIUtils.getStringArray(R.array.internet)
        }
    }


    fun getSuffix(position: Int, type: String): String {
        when (type) {
            Constant.ZHIHU -> return UIUtils.getStringArray(R.array.zhihu_suffix)[position]
            Constant.MOVIE -> return UIUtils.getStringArray(R.array.movie_suffix)[position]
            Constant.MUSIC -> return UIUtils.getStringArray(R.array.music_suffix)[position]
            Constant.DEVELOP -> return UIUtils.getStringArray(R.array.develop_suffix)[position]
            Constant.BOOK -> return UIUtils.getStringArray(R.array.book_suffix)[position]
            Constant.INTERNET -> return UIUtils.getStringArray(R.array.internet_suffix)[position]
            else -> return ""
        }
    }
}
